syft/pkg/github.go

```diff
@@ -0,0 +1,6 @@
+package pkg
+
+type GitHubActionsUseStatement struct {
```

Comment:
This seems to border "debug info" a little bit -- is it necessary? If Syft hasn't parsed a version or a sha, it's unlikely this is of use to anyone else, is it? I wonder if there would be utility in allowing different "levels" of detail in the SBOM. I know we have other catalogers doing similar stuff, so this isn't a blocker, just an opinion.

Reply:
It felt necessary since we're parsing a comment and treating it as the source of truth for a package version. The original version is lost without having a metadata type that exposes this.

> If Syft hasn't parsed a version or a sha, it's unlikely this is of use to anyone else, is it?

this resonates in the sense that if there is no comment to parse this is a useless metadata object.

> I wonder if there would be utility in allowing different "levels" of detail in the SBOM.

Do you have an example or idea how this would work?

Reply:
I don't really have an example but I would think something like logging levels but for SBOM detail. Maybe we determine level=1 is NTIA minimum elements, and level=2, 3, etc.. add more details up to including parsed data like full MANFIEST.MF and Portable Executable properties, etc.. where the lowest level excludes these details. There probably isn't a reason to do this, just an idle thought. This all was me thinking about SBOM size that some people may at some point be concerned with this.